### PR TITLE
feat(auth): convert downloadSubscription email templates

### PR DESCRIPTION
**Pull request by dschom**
Because:

* We are actively converting old FxA emails to our new modernized stack

This commit:

* Converts the `downloadSubscription` subscription platform email to the new stack.

Closes: #9267  

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots
New:
image

Old:
image

### CODE REVIEW

File: packages/fxa-auth-server/test/local/senders/mjml-emails.ts [hunk 1]
```diff
@@ -1036,6 +1037,29 @@ const TESTS: [string, any, Record<string, any>?][] = [
...
+    ['subject', { test: 'equal', expected: `Welcome to ${MESSAGE.productName}` }],
+    ['headers', new Map([
+      ['X-SES-MESSAGE-TAGS', { test: 'equal', expected: sesMessageTagsHeaderValue('downloadSubscription') }],
+      ['X-Template-Name', { test: 'equal', expected: 'downloadSubscription' }],
+      ['X-Template-Version', { test: 'equal', expected: TEMPLATE_VERSIONS.downloadSubscription }],
+    ])],
+    ['html', [
+      { test: 'include', expected: configHref('subscriptionTermsUrl', 'new-subscription', 'subscription-terms') },
+      { test: 'include', expected: decodeUrl(configHref('subscriptionSettingsUrl', 'new-subscription', 'cancel-subscription', 'plan_id', 'product_id', 'uid', 'email')) },
+      { test: 'include', expected: decodeUrl(`${MESSAGE.link}`) },
```
Comment (xlisachan) [inline packages/fxa-auth-server/test/local/senders/mjml-emails.ts:1050]: What happened to the test for `subscriptionSupportUrl`?

File: packages/fxa-auth-server/test/local/senders/mjml-emails.ts [hunk 2]
```diff
@@ -1036,6 +1037,29 @@ const TESTS: [string, any, Record<string, any>?][] = [
...
+      { test: 'include', expected: decodeUrl(`${MESSAGE.appStoreLink}`) },
+      { test: 'include', expected: `If you haven’t already downloaded ${MESSAGE.productName}, let’s get started using all the features included in your subscription:` },
+      { test: 'notInclude', expected: 'utm_source=email' },
+    ]],
+    ['text', [
+      { test: 'include', expected: `Welcome to ${MESSAGE.productName}` },
```
Comment (xlisachan) [inline packages/fxa-auth-server/test/local/senders/mjml-emails.ts:1056]: `text` is missing lines of tests from `email.js` as well...

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 3]
```diff
@@ -1,38 +1,64 @@
...
+<%# This Source Code Form is subject to the terms of the Mozilla Public 
+  # License, v. 2.0. If a copy of the MPL was not distributed with this 
   # file, You can obtain one at http://mozilla.org/MPL/2.0/. %>
 
 <mj-include path="./css/appBadges/index.css" type="css" css-inline="inline" />
 
 <mj-section>
-  <mj-group>
+  <mj-group css-class="app-badges">
+    <% if (locals.iosUrl) { %>
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:9]: @LZoog , tapping you in here, I don't think we should have this?

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 4]
```diff
@@ -1,38 +1,64 @@
...
+        alt="android badge"
+      >
       </mj-image>
     </mj-column>
+    <% } %>
   </mj-group>
 </mj-section>
 
+<% if (!locals.hideDeviceLink) { %>
 <mj-section>
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:35]: I think if you make the condition in the class similar to line 36, then you don't have to add `hideDeviceLink` in `fxa-auth-server/lib/senders/email.js`...

**Review by xlisachan — COMMENTED** [review]
I'm going to work on the button partial so it can be included in this PR.

Other than updating the tests, I'm aligned with the email texts, but I need @LZoog to confirm the stories/using parts of the `appBadges`. 🤔

File: packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.stories.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { Meta } from '@storybook/html';
+import { subplatStoryWithProps } from '../../storybook-email';
+
+export default {
+  title: 'SubPlat Emails/Templates/downloadSubscription',
+} as Meta;
+
+const createStory = subplatStoryWithProps(
+  'downloadSubscription',
+  'Sent when a user adds a subscription.',
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.stories.ts:14]: Small nit: we could use the word "successfully" since if their payment fails they're sent something else.

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 6]
```diff
@@ -1,33 +1,44 @@
...
+<% locals.iosUrl = locals.iosUrl || locals.appStoreLink %> 
+<% locals.androidUrl = locals.androidUrl ||  locals.playStoreLink %> 
+
 <mj-section>
-  <mj-group>
+  <mj-group css-class="app-badges">
     <mj-column>
-      <mj-image css-class="app-badges"
+      <mj-image
+        css-class="app-badge"
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:15]: I think you can revert the changes here and take out the `app-badge` class too.

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.scss [hunk 7]
```diff
@@ -92,3 +92,10 @@
 .text-footer-automatedEmail {
   @extend .text-footer;
 }
+
+.text-footer-appBadges div {
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.scss:96]: Do we need this class and div at all if we shouldn't render that "install on a desktop device" text? I think we can remove it. 🤔

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.scss [hunk 8]
```diff
@@ -3,9 +3,13 @@
...
 
 @use '../../global.scss';
-@use '../../layouts/fxa/index.scss';
 
 .app-badges {
+  max-width: 310px !important;
+  text-align: center !important;
+}
+
+.app-badge {
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.scss:12]: Hmm, since the text on Subplat is larger, it looks to me like we might want a little bit more space underneath these to give some breathing room/clickable space between the badges and the "contact us" link:

image


Though, that might require some style shuffling around in layouts, or adding a conditional class here or on `subscriptionSupport`. This is definitely not a blocker.

File: packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.mjml [hunk 9]
```diff
@@ -0,0 +1,60 @@
+<%# This Source Code Form is subject to the terms of the Mozilla Public 
+  # License, v. 2.0. If a copy of the MPL was not distributed with this 
+  # file, You can obtain one at http://mozilla.org/MPL/2.0/. %>
+
+<mj-include path="./css/subscription/index.css" type="css" css-inline="inline" />
+<mj-include path="./css/button/index.css" type="css" css-inline="inline" />
```
[the remark this answers is not in the document]
Comment (LZoog) [reply packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.mjml:5]: We can take a look whenever we look at that "click here" issue 👍

File: packages/fxa-auth-server/test/local/senders/mjml-emails.ts [hunk 10]
```diff
@@ -1421,7 +1454,7 @@ const TESTS: [string, any, Record<string, any>?][] = [
   ])],
 ];
 
-describe('lib/senders/mjml-emails:', () => {
+describe.only('lib/senders/mjml-emails:', () => {
```
Comment (LZoog) [inline packages/fxa-auth-server/test/local/senders/mjml-emails.ts:1457]: I think this disables the other tests right? Why do we need this here?

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 11]
```diff
@@ -1,33 +1,44 @@
...
-  # License, v. 2.0. If a copy of the MPL was not distributed with this
+<%# This Source Code Form is subject to the terms of the Mozilla Public 
+  # License, v. 2.0. If a copy of the MPL was not distributed with this 
   # file, You can obtain one at http://mozilla.org/MPL/2.0/. %>
 
 <mj-include path="./css/appBadges/index.css" type="css" css-inline="inline" />
 
+<%# TODO: Conslidate app store link names %>
+<% locals.iosUrl = locals.iosUrl || locals.appStoreLink %> 
+<% locals.androidUrl = locals.androidUrl ||  locals.playStoreLink %>
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:9]: Yeah I am now no longer confident that we will always have both of these when users download a subscription product. 😅 I think it's all fine to leave the variables like this (but you could do `iosUrl = locals.iosUrl | locals.appStoreLink`), but we can just have a conditional to show the app badge like how you had before, `<% if (iosURL) { %> // image` and also include the storybooks to show one or the other, I think that'd be awesome. Sorry to ask you to change something I suggested you change in the first place.

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 12]
```diff
@@ -1,32 +1,54 @@
...
+
   </mj-group>
 </mj-section>
 
+<% if (!locals.hideDeviceLink) { %>
 <mj-section>
   <mj-column>
     <% if (locals.onDesktopOrTabletDevice) { %>
-      <mj-text css-class="text-footer-appBadges"><span data-l10n-id="another-desktop-device"> Or, install on
+       <mj-text css-class="text-footer-appBadges"><span data-l10n-id="another-desktop-device"> Or, install on
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:51]: Nitipick: extra space in the indentation.

File: packages/fxa-auth-server/lib/senders/emails/partials/subscriptionSupport/index.mjml [hunk 13]
```diff
@@ -3,7 +3,7 @@
   # file, You can obtain one at http://mozilla.org/MPL/2.0/. %>
 
 <mj-section>
-  <mj-column>
+    <mj-column css-class="<%= locals.cssClass || undefined %>">
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/partials/subscriptionSupport/index.mjml:6]: Nitpick: extra space in indentation.

**Review by xlisachan — APPROVED** [review]
r+! Great job, @dschom! Thanks for taking this email template! 🎉

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 14]
```diff
@@ -1,38 +1,61 @@
...
       </mj-image>
     </mj-column>
+    <% } %>
+
+    <%# Ensure bottom margin is retained. %>
+    <% if (!locals.iosUrl && !locals.androidUrl) { %>
+    <mj-column css-class="text-body">
+      <mj-text></mj-text>
+    </mj-column>
+    <% } %>
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:42]: Instead of an extra empty DOM element, can we conditionally set a margin-bottom class?

I think we can do something like `<mj-group css-class="app-badges <%= !locals.iosUrl && !locals.androidUrl && 'mb-5' %>">` (not tested)

File: packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml [hunk 15]
```diff
@@ -1,38 +1,61 @@
...
 </mj-section>
 
+<% if (!locals.hideDeviceLink) { %>
 <mj-section>
   <mj-column>
     <% if (locals.onDesktopOrTabletDevice) { %>
-      <mj-text css-class="text-footer-appBadges"><span data-l10n-id="another-desktop-device"> Or, install on
-        <a href="<%- desktopLink %>" class="link-blue" data-l10n-name="anotherDeviceLink">another desktop device</a></span>
-      </mj-text>
+    <mj-text css-class="text-footer-appBadges"><span data-l10n-id="another-desktop-device"> Or, install on
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/partials/appBadges/index.mjml:51]: Another small nit, the spacing here was probably "more correct" before and this is missing an empty line at the end of the file.

My two comments so far aren't blockers though, it's fine in a quick follow up.

File: packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.stories.ts [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+    playStoreLink: undefined,
+  },
+  'Just Apple App Store'
+);
+
+export const DownloadSubscriptionJustAndroidStore = createStory(
+  {
+    appStoreLink: undefined,
+  },
+  'Just Andriod App Store'
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.stories.ts:43]: Also a typo here.

File: packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.stories.ts [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    link: 'http://getfirefox.com/',
+    productName: 'Firefox Fortress',
+    subscriptionSupportUrl: 'http://localhost:3030/support',
+    playStoreLink:
+      'https://play.google.com/store/apps/details?id=org.mozilla.firefox',
+    appStoreLink:
+      'https://apps.apple.com/us/app/firefox-private-safe-browser/id989804926',
+    // Forces fallbacks to be used in appBadges partial
+    iosUrl: undefined,
+    androidUrl: undefined,
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/templates/downloadSubscription/index.stories.ts:26]: https://github.com/mozilla/fxa/blob/ca61a3fac9d691676e4a6ce5b41c5753cd8f174f/packages/fxa-auth-server/lib/senders/emails/storybook-email.ts#L27

I don't know that we need to explicitly set these to `undefined` but it's OK to leave this as-is.

Sorry for not starting a review, this is my last comment. Thank you for all your work here, this is all super great!! 🙂

**Review by LZoog — APPROVED** [review]
